src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -84,7 +84,7 @@ let mkLdfldMethodDef (ilMethName, reprAccess, isStatic, ilTy, ilFieldName, ilPro
 
             mkILNonGenericInstanceMethod (ilMethName, reprAccess, [], ilReturn, body)
 
-    ilMethodDef.WithSpecialName
+    ilMethodDef.With(customAttrs = mkILCustomAttrs customAttrs).WithSpecialName
```

Comment:
Maybe create WithCustomAttributes similar to WithSpecialName?

Reply:
That method should *probably* include merging with whatever attributes may already be on the method def. Whereas here we know there won't be any so we can skip that. Also in other places custom attributes are added the same way already.

Reply:
Alright